test_db.py

```diff
@@ -0,0 +1,231 @@
+import unittest
+import uuid
+from PIL import Image
+import io
+import base64
+import db.queries.user as user
+import db.queries.seed as seed
+import db.queries.picture as picture
+import metadata.picture_set as picture_set_data
+import metadata.picture as picture_data
+import db.__init__ as db
+
+"""
+This is a test script for the database packages. 
+It tests the functions in the user, seed and picture modules.
+"""
+
+
+def is_valid_uuid(val):
+    """
+    This validates if a given string is a UUID
+    """
+    try:
+        uuid.UUID(str(val))
+        return True
+    except ValueError:
+        return False
+
+
+# --------------------  USER FUNCTIONS --------------------
+class test_user_functions(unittest.TestCase):
+    def setUp(self):
+        self.con = db.connect_db()
+        self.cursor = self.con.cursor()
+        self.email = "test@email.gouv.ca"
+
+    def tearDown(self):
+        db.end_query(self.con, self.cursor)
+
+    def test_is_user_registered(self):
+        """
+        This test checks if the is_user_registered function returns the correct value
+        for a user that is not yet registered and one that is.
+        """
+
+        self.assertFalse(
+            user.is_user_registered(self.cursor, self.email),
+            "The user should not already be registered",
+        )
+
+        user_id = user.register_user(self.cursor, self.email)
+
+        self.assertTrue(is_valid_uuid(user_id), "The user_id is not a valid UUID")
+        self.con.rollback()
+
+    def test_get_user_id(self):
+        """
+        This test checks if the get_user_id function returns the correct UUID
+        """
+        user_id = user.register_user(self.cursor, self.email)
+        uuid = user.get_user_id(self.cursor, self.email)
+
+        self.assertTrue(is_valid_uuid(user_id), "The user_id is not a valid UUID")
+        self.assertTrue(is_valid_uuid(uuid), "The returned UUID is not a valid UUID")
+        self.assertEqual(
+            user_id, uuid, "The returned UUID is not the same as the registered UUID"
+        )
+
+        self.con.rollback()
```

Comment:
instead of assertTrue, you can either use `assertEqual` to validate the uuid value or `assertIsInstance` to validate the type of the uuid

Reply:
like stated in other comments, the uuid are under a string type. I want to validate the string is a uuid.